src/librustc_trans/mir/rvalue.rs

```diff
@@ -478,8 +478,10 @@ impl<'a, 'tcx> MirContext<'a, 'tcx> {
             }
             mir::Rvalue::Repeat(..) |
             mir::Rvalue::Aggregate(..) => {
-                bug!("cannot generate operand from rvalue {:?}", rvalue);
-
+                // According to `rvalue_creates_operand`, only ZST
+                // aggregate rvalues are allowed to be operands.
```

Comment:
(debug) assertion would be nice here. It is not unlikely for somebody to edit some interaction between some `rvalue_creates_operand` and this function in a wrong way and an assertion here would catch it.